src/zero_copy/primitive.rs

```diff
@@ -404,6 +404,63 @@ where
     go_extra!();
 }
 
+pub struct FilterMapState<F, E> {
```

Comment:
I agree that this is a useful combinator, but the name conflict with the existing `filter` and `filter_map` primitives make me uncomfortable. That said, I think that it's those primitives that were named in error, and not this combinator. What do you think about removing those primitives entirely and replacing them with `any().filter(..)`/`any().filter_map(...)`?

Reply:
That sounds good to me. It seems kind of odd that you 'filter' as a first step. I feel like I've definitely reached for filter as a trait method and had to remind myself it wasn't at least once or twice.

Reply:
With that, `try_map_state` might be a better name, at least for my use case

Reply:
Perhaps `try_map_with_state`, to mirror `map_with_span`?

Reply:
That works. Also feel like this is hitting the std situation: params x, y, and z, do we add all the superset of combinations, or how do we pick which should be included :P